x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/partition_missing_fields_events.ts

```diff
@@ -15,16 +16,20 @@ import type { SignalSourceHit } from '../types';
  * 1. first one, where no suppressed by field has empty value
  * 2. where any of fields is empty
  */
-export const partitionMissingFieldsEvents = (
-  events: SignalSourceHit[],
-  suppressedBy: string[] = []
-): SignalSourceHit[][] => {
+export const partitionMissingFieldsEvents = <
+  T extends SignalSourceHit | { event: SignalSourceHit }
+>(
+  events: T[],
+  suppressedBy: string[] = [],
+  fieldsPath: ['event'] | [] = []
```

Comment:
This parameter is pretty confusing, and a lack of tests/documentation doesn't help. Why is this an array (as opposed to a dotted string, etc)? Why is `['event']` the only valid value?

Reply:
added comment and test.
event only valid value for now as fields property can be located at `event` key for new terms rule.
Which operates with object of event and new term, instead of just event as the rest of the rules